server.js

```diff
@@ -23,50 +23,48 @@ app.use(cookieParser());
 
 // get requests
 app.get("/", async (req, res) => {
-    const { cookies } = req
     try {
-        const currentSession = await Session.findById(cookies.session_id)
-        if (currentSession === null) {
-            res.render('index', currentSession);
-        } else {
-            res.render('index', currentSession);
+        if (!req.cookies.session) {
+            res.render('index');
+            return;
         }
-
+        console.log(req.cookies.session);
+        const currentSession = await Session.findById(req.cookies.session);
+        if (!currentSession) {
+            res.clearCookie('session');
+            res.render('index');
+            return;
+        }
+        const currentUser = await User.findById(currentSession.user);
+        if (!currentUser) {
+            res.clearCookie('session');
+            res.render('index');
+            return
+        }
+        res.redirect('/profile');
     } catch (err) {
         res.render('error', { "error message": err })
     }
 })
 
-app.get("/profile/:id", async (req, res) => {
-    const { cookies } = req
+app.get("/profile", async (req, res) => {
     try {
-        const currentSession = await Session.findById(cookies.session_id)
-        const curentuser = await User.findOne({username: currentSession.user})
-        console.log(curentuser);
-        console.log(curentuser.id);
-        console.log(currentSession.id);
-        if (req.params.id !== curentuser.id) {
-            res.redirect(`/profile/${curentuser.id}`)
-        } else {
-            if (cookies.session_id) {
-                res.render("profile", currentSession)
-            } else {
-                res.redirect('/')
-            }
+        if (!req.cookies.session) {
+            res.redirect('/')
+            return;
         }
-
+        const currentSession = await Session.findById(req.cookies.session);
+        const currentUser = await User.findById(currentSession.user);
```

Comment:
What happens if I do have a session, but it's not in the database? Remember, the client can send you whatever it wants.